contrib/tokenmgmr

```diff
@@ -0,0 +1,146 @@
+development experience. If there is some missing prerequisite for running the
+functional tests (like an API token), then we simply log an error and then rely
+on the Pytest implementation to gracefully skip the tests being run.
+
+"""
+
+from __future__ import print_function
+
+import click
+import os
+import sh
+
+from molecule import logger
+
+log = logger.get_logger(__name__)
+
+IS_TRAVIS = os.getenv('TRAVIS') and os.getenv('CI')
+
+
+def _ensure_hetzner_cloud_tts_token():
+    """Check that a Hetzner Cloud TTS token is exposed."""
+    if not os.getenv('HETZNER_CLOUD_TTS_TOKEN'):
+        log.error((
+            'No Hetzner Cloud TTS token available. Please '
+            'refer to the CI service configuration which should '
+            'be configured to expose this environment variable.'
+        ))
+
+
+def _ensure_hetzner_cloud_api_token():
+    """Check that a Hetzner Cloud API token is exposed."""
+    if not os.getenv('HCLOUD_TOKEN'):
+        log.error((
+            'No Hetzner Cloud API token '
+            'environment variable exposed'
+        ))
+
+
+def _get_hetzner_cloud_token():
+    """Retrieve a Hetzner Cloud API token."""
+    if not IS_TRAVIS:
+        _ensure_hetzner_cloud_api_token()
+        return os.getenv('HCLOUD_TOKEN')
+
+    _ensure_hetzner_cloud_tts_token()
+
+    try:
+        return sh.curl(
+            'https://tt-service.hetzner.cloud/token',
+            A='travis-ansible-provider',
+            H='Authorization: Bearer ${HETZNER_CLOUD_TTS_TOKEN}',
+            X='POST',
+        )
+    except sh.ErrorReturnCode as exception:
+        log.error((
+            'Unable to retrieve Hetzner Cloud API token '
+            'via a curl invocation. Saw the following '
+            'error:\n{}'
+        ).format(str(exception)))
+
+
+def _remove_hetzner_cloud_token():
+    """Remove a Hetzner Cloud API token."""
+    if not IS_TRAVIS:
+        return
+
+    _ensure_hetzner_cloud_tts_token()
+    _ensure_hetzner_cloud_api_token()
+
+    try:
+        token = os.getenv('HCLOUD_TOKEN')
+        return sh.curl(
+            'https://tt-service.hetzner.cloud/token?token={}'.format(token),
+            A='travis-ansible-provider',
+            H='Authorization: Bearer ${HETZNER_CLOUD_TTS_TOKEN}',
+            X='DELETE',
+        )
+    except sh.ErrorReturnCode as exception:
+        log.error((
+            'Unable to remove a Hetzner Cloud API token '
+            'via a curl invocation. Saw the following '
+            'error:\n{}'
+        ).format(str(exception)))
+
+
+@click.group()
+def main():
+    """Retrieve tokens for interacting with driver providers."""
+    pass
+
+
+@main.command()
+@click.option(
+    '--driver-name',
+    '-d',
+    type=click.Choice(['hetznercloud']),
+    help='Name of driver to get token for.')
+def get(driver_name):
+    """Retrieve a temporary CI token."""
+    if driver_name == 'hetznercloud':
+        return print(_get_hetzner_cloud_token())
```

Comment:
This will return `None`. If that's what you expect, the return statement is completely unnecessary.
```suggestion
        print(_get_hetzner_cloud_token())
```
Also, I'm pretty sure that you should use `click.echo` instead.